src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,96 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int EMPTY_SIZE = 0;
+    private static final int CAPACITY_DIVISOR = 2;
+    private Object[] dataElement = new Object[DEFAULT_CAPACITY];
+    private int size;
+
     @Override
     public void add(T value) {
-
+        grow();
+        dataElement[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
 
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Can't add to index " + index);
+        }
+        grow();
+        System.arraycopy(dataElement, index, dataElement, index + 1, size++ - index);
+        dataElement[index] = value;
     }
 
     @Override
     public void addAll(List<T> list) {
 
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndexBounds(index);
+        return (T) dataElement[index];
```

Comment:
```suggestion
        return  dataElement[index];
```
You don't need to cast directly.
Fix the relevant errors related to this